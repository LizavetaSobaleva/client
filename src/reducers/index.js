import { applyMiddleware, combineReducers } from "redux";
import { configureStore} from "@reduxjs/toolkit"
import { thunk } from "redux-thunk";
import userReducer from "./userReducer";
import fileReducer from "./fileReducer";
import uploadReducer from "./uploadReducer";
import appReducer from "./appReducer";
import { composeWithDevTools } from '@redux-devtools/extension';


const rootReducer = combineReducers({
    user: userReducer,
    files: fileReducer,
    upload: uploadReducer,
    app: appReducer
})

export const store = configureStore({reducer: rootReducer, devTools: composeWithDevTools(applyMiddleware(thunk))})