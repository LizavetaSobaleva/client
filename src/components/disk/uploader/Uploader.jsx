import React from 'react'
import './uploader.less'
import Popup from '../../UI/popup/Popup'
import UploadFile from './UploadFile'
import { useDispatch, useSelector } from 'react-redux'
import { hideUploader } from '../../../reducers/uploadReducer'

const Uploader = () => {
    const files = useSelector(state => state.upload.files)
    const isVisible = useSelector(state => state.upload.isVisible)
    const dispatch = useDispatch()

  return (
    isVisible &&
    <div className='uploader'>
        <Popup header={'Uploaded files'} onClick={() => dispatch(hideUploader())}>
            <div className="uploader__content">
                {files.map(file => <UploadFile key={file.id} file={file} />)}
            </div>
        </Popup>
    </div>
  )
}

export default Uploader