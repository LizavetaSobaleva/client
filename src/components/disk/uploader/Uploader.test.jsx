import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Uploader from './Uploader';
import { hideUploader } from '../../../reducers/uploadReducer';


jest.mock('./UploadFile', () => ({ file }) => (
  <div data-testid={`file-${file.id}`}>{file.name}</div>
));

// Mocking Popup component
jest.mock('../../UI/popup/Popup', () => ({ header, onClick, children }) => (
  <div className='popup'>
    <div className="popup__header">
      <div className="popup__title" data-testid="popupTitle">{header}</div>
      <button onClick={onClick} data-testid="popupCloseBtn">Close</button>
    </div>
    <div>{children}</div>
  </div>
));

const mockStore = configureStore([]);

const files = [
  { id: '1', name: 'file1.txt', progress: 50 },
  { id: '2', name: 'file2.jpg', progress: 75 },
];

let store;

beforeEach(() => {
  store = mockStore({
    upload: {
      files,
      isVisible: true,
    },
  });
  store.dispatch = jest.fn();
});

const renderComponent = () => render(
  <Provider store={store}>
    <Uploader />
  </Provider>
);

test('renders Uploader without crashing', () => {
  renderComponent();
  expect(screen.getByTestId('popupTitle')).toHaveTextContent('Uploaded files');
  files.forEach(file => {
    expect(screen.getByTestId(`file-${file.id}`)).toBeInTheDocument();
  });
});

test('hides Uploader when close button is clicked', () => {
  renderComponent();
  fireEvent.click(screen.getByTestId('popupCloseBtn'));
  expect(store.dispatch).toHaveBeenCalledWith(hideUploader());
});

test('does not render Uploader when isVisible is false', () => {
  store = mockStore({
    upload: {
      files,
      isVisible: false,
    },
  });
  renderComponent();
  expect(screen.queryByTestId('popupTitle')).not.toBeInTheDocument();
});
