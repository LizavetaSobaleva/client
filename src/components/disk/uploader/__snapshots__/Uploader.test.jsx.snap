// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides Uploader when close button is clicked 1`] = `
<DocumentFragment>
  <div
    class="uploader"
  >
    <div
      class="popup"
    >
      <div
        class="popup__header"
      >
        <div
          class="popup__title"
          data-testid="popupTitle"
        >
          Uploaded files
        </div>
        <button
          data-testid="popupCloseBtn"
        >
          Close
        </button>
      </div>
      <div>
        <div
          class="uploader__content"
        >
          <div
            data-testid="file-1"
          >
            file1.txt
          </div>
          <div
            data-testid="file-2"
          >
            file2.jpg
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders Uploader without crashing 1`] = `
<DocumentFragment>
  <div
    class="uploader"
  >
    <div
      class="popup"
    >
      <div
        class="popup__header"
      >
        <div
          class="popup__title"
          data-testid="popupTitle"
        >
          Uploaded files
        </div>
        <button
          data-testid="popupCloseBtn"
        >
          Close
        </button>
      </div>
      <div>
        <div
          class="uploader__content"
        >
          <div
            data-testid="file-1"
          >
            file1.txt
          </div>
          <div
            data-testid="file-2"
          >
            file2.jpg
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
